data_structures/binary_tree/binary_tree_traversals.py

```diff
@@ -1,24 +1,16 @@
 # https://en.wikipedia.org/wiki/Tree_traversal
+from dataclasses import dataclass
 
 
+@dataclass
 class Node:
-    """
-    A Node has data variable and pointers to its left and right nodes.
-    """
-
-    def __init__(self, data):
-        self.left = None
-        self.right = None
-        self.data = data
+    data: int
+    left: "Node" = None
+    right: "Node" = None

```

Comment:
I did not notice this the first time but shouldn't this be `Optional["Node"]`?

Reply:
My sense was that if neither Python nor mypy was complaining about it then I would not either.  This would be the only location where we would set these values to `None` unless we started deleting Nodes which I assumed we were not gonna do.

Reply:
Python is never going to complain as it simply does not care about type hints. `mypy` will complain when you run it in a bit stricter modes. Plain `mypy` will not complain about a lot of stuff as that's how gradual typing goes.

And no, as per this definition, the leaves of the trees (the last nodes) will have its left and right node `None`

Reply:
> This would be the only location where we would set these values

Reply:
Yes, but that still won't allow the file to be type checked later on. The variable's correct type is `Optional["Node"]`. Just because it is being set `None` once, does not change the type of that variable. That's what 'static' means in static typing :)